File: sample/src/main/java/cc/coopersoft/android/keycloak/SplashActivity.java
package cc.coopersoft.android.keycloak;

import android.os.Handler;
import android.os.Looper;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;

public class SplashActivity extends AppCompatActivity {

  private Handler mHandler = new Handler(Looper.getMainLooper());

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_splash);
    mHandler.postDelayed(this::start, 3000);
  }


  private void start(){
    if (KeycloakClient.accessTokenHolder().hasToken()){
      MainActivity.start(this);
    }else{
      LoginActivity.start(this);
    }
  }
}

File: sample/src/main/java/cc/coopersoft/android/keycloak/SplashActivity.kt
package cc.coopersoft.android.keycloak

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.appcompat.app.AppCompatActivity
import cc.coopersoft.android.keycloak.KeycloakClient.accessTokenHolder

class SplashActivity : AppCompatActivity() {
    private val mHandler = Handler(Looper.getMainLooper())
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        mHandler.postDelayed({ this.start() }, 3000)
    }

    private fun start() {
        if (accessTokenHolder()!!.hasToken()) {
            MainActivity.start(this)
        } else {
            LoginActivity.start(this)
        }
    }
}